lalrpop/src/normalize/lower/mod.rs

```diff
@@ -349,7 +352,15 @@ impl<'s> LowerState<'s> {
                 if nt_type.is_unit() {
                     "()".to_string()
                 } else {
-                    "(<>)".to_string()
+                    let len = match &normalized_symbols {
+                        Symbols::Named(names) => names.len(),
+                        Symbols::Anon(indices) => indices.len(),
+                    };
+                    if len == 1 {
+                        "<>".to_string()
+                    } else {
+                        "(<>)".to_string()
+                    }
```

Comment:
nice! I was just trying to debug this warning and realised you've already fixed it :) this PR is really educational, thanks!